backend/drm/backend.c

```diff
@@ -93,7 +93,13 @@ static void session_signal(struct wl_listener *listener, void *data) {
 		struct wlr_drm_connector *conn;
 		wl_list_for_each(conn, &drm->outputs, link){
 			if (conn->output.enabled) {
-				wlr_output_set_mode(&conn->output, conn->output.current_mode);
+				// wlr_output_set_mode prevents setting the mode to the
+				// current mode since it is almost always a noop. This however
+				// is one exception so just set current_mode to NULL before
+				// setting the mode.
+				struct wlr_output_mode *mode = conn->output.current_mode;
+				conn->output.current_mode = NULL;
+				wlr_output_set_mode(&conn->output, mode);
```

Comment:
Can't we just call our DRM-private function in `drm/drm.c`?

Reply:
Yup. Just had to change the scope of it to so it was no longer static.